tgui/packages/tgui/interfaces/Pandemic.tsx

```diff
@@ -0,0 +1,484 @@
+          <LabeledList.Item label="DNA">
+            {blood.dna.replace(/^\w/, (c) => c.toUpperCase())}
+          </LabeledList.Item>
+          <LabeledList.Item label="Type">
+            {blood.type.replace(/^\w/, (c) => c.toUpperCase())}
+          </LabeledList.Item>
+        </LabeledList>
+      </Stack.Item>
+      <Stack.Item grow={2}>
+        <LabeledList>
+          <LabeledList.Item label="Container">
+            <ProgressBar
+              color="darkred"
+              value={beaker.volume}
+              minValue={0}
+              maxValue={beaker.capacity}
+              ranges={{
+                'good': [beaker.capacity * 0.85, beaker.capacity],
+                'average': [beaker.capacity * 0.25, beaker.capacity * 0.85],
+                'bad': [0, beaker.capacity * 0.25],
+              }}
+            />
+          </LabeledList.Item>
+        </LabeledList>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+/** If antibodies are present, returns buttons to create vaccines */
+const AntibodyInfoDisplay = (_, context) => {
+  const { act, data } = useBackend<PandemicContext>(context);
+  const { is_ready, resistances } = data;
+  if (!resistances) {
+    return <NoticeBox>Nothing detected</NoticeBox>;
+  }
+
+  return (
+    <LabeledList>
+      <LabeledList.Item label="Antibodies">
+        {!resistances.length
+          ? 'None'
+          : resistances.map((resistance) => {
+            return (
+              <Button
+                key={resistance.name}
+                icon="eye-dropper"
+                disabled={!is_ready}
+                tooltip="Creates a vaccine bottle."
+                onClick={() =>
+                  act('create_vaccine_bottle', {
+                    index: resistance.id,
+                  })}>
+                {`${resistance.name}`}
+              </Button>
+            );
+          })}
+      </LabeledList.Item>
+    </LabeledList>
+  );
+};
+
+/** Displays info for the loaded blood, if any */
+const SpecimenDisplay = (_, context) => {
+  const { act, data } = useBackend<PandemicContext>(context);
+  const [tab, setTab] = useSharedState(context, 'tab', 0);
+  const { is_ready, viruses } = data;
+  if (!viruses?.length) {
+    return <NoticeBox>No viruses detected</NoticeBox>;
+  }
+  const virus = viruses[tab];
+
+  return (
+    <Section
+      fill
+      scrollable
+      title="Specimen"
+      buttons={
+        <Stack>
+          {
+            // Tabs if there's more viruses
+            viruses.length > 1 && (
+              <Stack.Item>
+                <Tabs>
+                  {viruses.map((virus, index) => {
+                    return (
+                      <Tabs.Tab
+                        selected={tab === index}
+                        onClick={() => setTab(index)}
+                        key={virus.name}>
+                        {virus.name}
+                      </Tabs.Tab>
+                    );
+                  })}
+                </Tabs>
```

Comment:
This is starting to get a little indented and can probably be made into its own component for better readability.

Reply:
Done